src/Native/System.Native/pal_process.cpp

```diff
@@ -59,6 +65,9 @@ static_assert(PAL_PRIO_PROCESS == static_cast<int>(PRIO_PROCESS), "");
 static_assert(PAL_PRIO_PGRP == static_cast<int>(PRIO_PGRP), "");
 static_assert(PAL_PRIO_USER == static_cast<int>(PRIO_USER), "");
 
+// We make assumptions based on these being the same
+static_assert(sizeof(char) == sizeof(uint8_t), "");
```

Comment:
Can't hurt to be paranoid, I guess, but this is guaranteed.

http://en.cppreference.com/w/cpp/language/types
>Besides the minimal bit counts, the C++ Standard guarantees that 
>1 == sizeof(char) <= sizeof(short) <= sizeof(int) <= sizeof(long) <= sizeof(long long).